### PR TITLE
feat: deploy service to shuttle.dev

### LINKED ISSUE #38: Deploy Service to Shuttle.dev with CI/CD Pipeline
**Description**  
Deploy this `gRPC` service to [[Shuttle.dev](https://www.shuttle.dev/)](https://www.shuttle.dev/) and set up a CI/CD pipeline using the [[GitHub Workflow template](https://github.com/tailcallhq/gh-workflow)](https://github.com/tailcallhq/gh-workflow).  

**Tasks**  
1. **Set Up Shuttle.dev**  
   - Configure the project for deployment on Shuttle.dev.  
   - Test deployment locally using `cargo shuttle run`.  

2. **CI/CD Pipeline Integration**  
   - Add the CI/CD pipeline using the [[tailcallhq GitHub Workflow](https://github.com/tailcallhq/gh-workflow)](https://github.com/tailcallhq/gh-workflow).  
   - Ensure the pipeline includes steps for:  
     - Linting (Rustfmt/Clippy).  
     - Testing (`cargo test`).  
     - Deployment to Shuttle.dev upon merging to the `main` branch.  

3. **Documentation**  
   - Update the project README with details about Shuttle.dev deployment and CI/CD pipeline.  
   - Include setup instructions for local testing and development.  

**Acceptance Criteria**  
- The service is deployed on Shuttle.dev and accessible publicly.  
- CI/CD pipeline successfully triggers and deploys on merges to the `main` branch.  
- Documentation is updated and clear.

### PR DESCRIPTION
**Pull request by neo773**
image

image

/claim #38
/closes #38

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,15 +15,34 @@ This repository contains a gRPC-based Rust server implementing CRUD operations f
...
 ```bash
-cargo run
+cargo shuttle run --port 50051
+```
+
+## Deploying to Shuttle.dev
+
+Deploy the server with:
+
+```bash
```
Comment (amitksingh1490) [inline README.md:34]: Do we need to do this step for every deployment instructions not very clear

File: README.md [hunk 2]
```diff
@@ -15,15 +15,34 @@ This repository contains a gRPC-based Rust server implementing CRUD operations f
...
+```
+
+Follow the prompts:
+1. Select your project name
+2. Choose 'Empty (no framework)' when prompted
+
+Then deploy with:
+
+```bash
+cargo shuttle deploy
```
Comment (amitksingh1490) [inline README.md:45]: does this require setting up keys on shuttle.dev?

Comment (neo773) [reply README.md:45]: Nope, CLI automatically logs in with browser OAuth

Comment (amitksingh1490) [reply README.md:45]: ya but If I need to run it on CI i would require some key based setup

File: tests/ci.rs [hunk 3]
```diff
@@ -40,18 +41,25 @@ fn generate() {
...
                 .add_branch("main"),
         );
 
-    let permissions = Permissions::default()
-        .pull_requests(Level::Write)
-        .packages(Level::Write)
-        .contents(Level::Write);
+    let deploy = Job::new("Deploy to Shuttle")
+        .permissions(Permissions::default().contents(Level::Write))
+        .cond(Expression::new("github.event.pull_request.merged == true"))
```
Comment (amitksingh1490) [inline tests/ci.rs:50]: use Context API for this and make it run on push to main Instead of merged pull request

Comment (amitksingh1490) [reply tests/ci.rs:50]: https://github.com/tailcallhq/gh-workflow/blob/11e5e82bbb2038b2bd1da67ee19a62d6585e8c7a/tests/ci.rs#L55

**Comment by neo773** [comment]
@amitksingh1490 
Added the Context API for deploy workflow

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**